### ISSUE TITLE
Vosk is dead. Long live vosk. Whisper is the new Vosk. Let's just parameterize it?

### ISSUE THREAD

**Issue by walt93** [WWBN/AVideo#8416]
**Is your feature request related to a problem? Please describe.**
Can we ust put the shell command in to a preference somewhere? It was vosk yesterday, today's flavor is whisper-ai (it leaves vosk in the dust, seriously), who knows what new AI tool tomorrow will bring.

**Describe the solution you'd like**
Similar to how we handle ffmpeg in the encoder, we could designate a variable like {videofile} and then we would just put the command in, vosk, whisper, mumbledeyfook whatever.

![Screen Shot 2023-09-11 at 7 47 58 PM](https://github.com/WWBN/AVideo/assets/11711050/dcb54a14-1c4c-4012-84e7-21e57921b549)

**Comment by DanielnetoDotCom** [comment]
Vosk is executed in the streamer, not in the encoder

what parameters do you want to add? it probably have to be added in the SubtitleSwitcher plugin

I am currently working on an AI plugin that will integrate Whisper. We are always keen on integrating cutting-edge technologies to enhance our platform.

Additionally, I'd like to point out that the OpenAI API isn't free, but its capabilities are impressive. To facilitate this, the usage of the OpenAI API through our AI plugin will consume the YPT CDN credits.

I am hopeful that this new feature will be available for release in the coming months.

**Comment by walt93** [comment]
I did a back of napkin calculation and my videos would cost several thousand dollars to run whisper on OpenAI servers. This maybe a good option for some.

We can run locally:

```
pip3 install whisper-ai
whisper --model base.en --output_format all --language Romanian "Romanian Senator Diane Sosoaca on HAARP.mp3"
```

You want it running on a GPU, which is probably not your main AVideo box. I've got a halfway finished system using RabbitMQ as a task broker that lets you queue whisper tasks on another box. Maybe when I get back to it, can do so in a way to integrate with AVideo?
